src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py

```diff
@@ -72,52 +73,51 @@ def _get_resource_group(self, **kwargs):
         return kwargs.get(self.resource_group_parameter_name)
 
 
-class AdvancedThreatProtectionTest(ScenarioTest):
-     
+class AdvancedThreatProtectionTest(ScenarioTest):   
+
     @AllowLargeResponse()
     @ResourceGroupPreparer(location=DEFAULT_LOCATION)
    # @ServerPreparer(engine_type='mysql', location=DEFAULT_LOCATION)
-    def test_mysql_advanced_threat_protection_mgmt(self, resource_group):
+
+    def test_mysql_advanced_threat_protection_mgmt(self, resource_group,):
         self._test_flexible_server_threat_model_update_mgmt('mysql', resource_group)
 
-    def _test_flexible_server_threat_model_update_mgmt(self, database_engine, resource_group):
-        
-        # Test to check if the defender is in enabled state and the provisioning state is succeeded
-        server = "Default"
-        defender_state = 'Enabled'
-        print("Hello world")
-        update_response = self.cmd('{} flexible-server threat-protection update -g {} -s {} --defender-state {} \
-                    '.format(database_engine, resource_group, server, defender_state)).get_output_in_json()
+    def _test_flexible_server_threat_model_update_mgmt(self, database_engine, resource_group):     
+        server="Default"
+        #Test to check if the defender is in enabled state and the provisioning state i succeeded
+        defender_state = mysql_flexibleservers.models.AdvancedThreatProtectionState.ENABLED.value or defender_state == mysql_flexibleservers.models.AdvancedThreatProtectionState.DISABLED.value
+        defender_name = mysql_flexibleservers.models.AdvancedThreatProtectionName.DEFAULT.value
+        update_response=self.cmd('{} flexible-server threat-protection update -g {} -s {} --defender-state {} -dn {}\
+                    '.format(database_engine, resource_group, server,defender_state,defender_name)).get_output_in_json()
         self.assertEqual(update_response['properties']['state'], defender_state)
         self.assertEqual(update_response['properties']['provisioningState'], 'Succeeded')
+        self.assertEqual(defender_name,"Default")
+        self.assertEqual(defender_state,"Enabled")
```

Comment:
the actual values here are not coming from the method to be tested. Why are checking the input values?